### PR TITLE
PCRE2: Explicitly specify cmake c standard

### PR DESCRIPTION
**Pull request by bryce-cahill**
* Fixes compilation issues with for loops

Specify library name and version:  **pcre2/10.42**

Fixes https://github.com/conan-io/conan-center-index/issues/19725

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/pcre2/all/conanfile.py [hunk 1]
```diff
@@ -98,6 +98,8 @@ def generate(self):
         tc.variables["PCRE2_BUILD_PCRE2_32"] = self.options.build_pcre2_32
         tc.variables["PCRE2_SUPPORT_JIT"] = self.options.support_jit
         tc.variables["PCRE2GREP_SUPPORT_CALLOUT_FORK"] = self.options.get_safe("grep_support_callout_fork", False)
+        # Fix for error: 'for' loop initial declarations are only allowed in C99 or C11 mode
+        tc.variables["CMAKE_C_STANDARD"] = "99"
```
Comment (SpaceIm) [inline recipes/pcre2/all/conanfile.py:102]: According to https://github.com/PCRE2Project/pcre2/pull/193, it's a recent change, so it shouldn't be forced globally like this regardless of pcre2 version. I would recommend to patch impacted versions instead.

Comment (bryce-cahill) [reply recipes/pcre2/all/conanfile.py]: Great point, thanks for the feedback :+1: 

I've replaced this with a patch for the 10.42 version instead.

Comment (bryce-cahill) [reply recipes/pcre2/all/conanfile.py]: Modified the patch so that it now cleanly applies to both 10.40 and 10.42

**Comment by RubenRBS** [comment]
Please remember to sign the CLA so that your contribution can be merged, thanks! :)

**Comment by bryce-cahill** [comment]
> Please remember to sign the CLA so that your contribution can be merged, thanks! :)

I've attempted to sign it twice but it appears to be stuck pending

**Review by SpaceIm — APPROVED** [review]

**Review by valgur — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**